relay-server/src/actors/store.rs

```diff
@@ -301,10 +301,13 @@ impl StoreService {
         let mut error = None;
 
         for bucket in buckets {
-            // Create a local bucket view to avoid splitting buckets unnecessarily. Since we produce
-            // each bucket separately, we only need to split buckets that exceed the size, but not
-            // batches.
-            for view in BucketsView::new(&[bucket]).by_size(batch_size).flatten() {
+            let mut remaining = Some(BucketView::new(&bucket));
```

Comment:
Introducing a new iterator just for this one usage didn't seem worth the abstraction. What do you think of this contraption?